### PR TITLE
feat(contrib): add badge uservice

### LINKED ISSUE #546: feat (contrib/uservices): github badges
see http://shields.io/ for specs

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

1. Description
1. Related issues
close #546
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: contrib/uservices/badge/src/schema.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+table! {
+    run (id) {
+        id -> Int8,
+        run_id -> Int8,
+        num -> Int8,
+        project_key -> Text,
+        workflow_name -> Text,
+        branch -> Nullable<Text>,
+        status -> Text,
```
Comment (cdumay) [inline contrib/uservices/badge/src/schema.rs]: You can use [diesel-derive-enum](https://crates.io/crates/diesel-derive-enum) to store status as enum

File: contrib/uservices/badge/src/configuration.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+#[derive(Default, Debug, Deserialize, Clone)]
+#[serde(default)]
+pub struct Configuration {
+    pub badge: BadgeConfiguration,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+#[serde(default)]
+pub struct BadgeConfiguration {
+    pub url: String,
```
Comment (cdumay) [inline contrib/uservices/badge/src/configuration.rs]: You should use [url::Url](https://crates.io/crates/url) with the feature **serde** to load and implicit check the given url

File: contrib/uservices/badge/src/configuration.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+pub struct KafkaConfiguration {
+    pub group: String,
+    pub user: String,
+    pub password: String,
+    pub broker: String,
+    pub topic: String,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+pub struct HTTPConfiguration {
```
Comment (cdumay) [inline contrib/uservices/badge/src/configuration.rs]: Can be replaced by [std::net::SocketAddr](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) which implement Serialize & Deserialize

Comment (bnjjj) [reply contrib/uservices/badge/src/configuration.rs]: http_server(state: WebState, http_bind: String, http_port: String) ======> http_server(state: WebState, addr: SocketAddr)
with format!("{}:{}", http_bind, http_port).into()

**Pull request merged**